src/consumer/frame.js

```diff
@@ -176,8 +176,8 @@ class Frame extends EventEmitter {
   * @param {object} event - The emitted message event.
   */
   handleProviderMessage(event) {
-    // 1. This isn't a JSONRPC message, exit.
-    if (!event.data.jsonrpc) return;
+    // 1. This isn't a JSONRPC message or iframe is null, exit.
```

Comment:
I think the appropriate fix is to remove the event listener altogether in cleanup()

Reply:
good catch! I've added `removeEventListener` in cleanup method. I'd also like to keep the iframe check here to avoid potential errors.

Reply:
good catch! I've added `removeEventListener` in cleanup method. I'd also like to keep the iframe check here to avoid potential errors.